### DISCUSSION TITLE
ArgList

### DISCUSSION THREAD

**Discussion by dashne** [Washi1337/AsmResolver#550] [Q&A] [answered]
how to check if a method parameter is ArgList (__argList) Type?

**Comment by BadRyuner** [comment]
```csharp
MethodDefinition somemethod = ... ;
var sig = somemethod.Signature;
var isHasArgList = sig.Attributes.HasFlag(AsmResolver.DotNet.Signatures.CallingConventionAttributes.VarArg);
// or
var isHasArgList = sig.CallingConvention == AsmResolver.DotNet.Signatures.CallingConventionAttributes.VarArg;
```

**Reply by dashne** [reply]
thank you for your reply, is that mean last parameter of method is arglist type

**Reply by Washi1337** [reply, marked as answer]
The `__arglist` parameter in C# does not actually translate to a physical parameter definition or type in the metadata/signature. `__arglist` is just C# syntax for the `vararg` calling convention, indicating there may be more arguments passed into the function beyond the ones defined in the method's signature.

See: https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGED2AbbEDGMAlpgHYAyRMECU2yANDCEdgD4ACATAIwCwAKEEcAzAAJuEngHZBAb0FilE8RwAsYgLIAKIqRhiAHgzEB9U1AQBzbEWQwAlIuUKBysQF9BzpaIkaAOW0nNxcfdy1tHhERaRNzSxs7GCiTLhMRBwcAbnCvAQ8gA
